www/~/%username/emails/modify.json.spt

```diff
@@ -46,7 +47,10 @@ if action in ('add-email', 'resend', 'start-verification'):
         raise Response(400)
 
     participant.start_email_verification(address, *packages)
-    msg = _("Check your inbox for a verification link.")
+    if show_address_in_message:
```

Comment:
> This is confusing 😕 

D'oh! 😞

> `verification.spt` is used both to verify emails and to verify linking packages to emails.

Yeah, it seemed to me that it was best to have one code path for both, since we want linking packages to emails to _also_ verify the email. That is, we don't want to make the user go through two verification flows (one for the address itself, a second for the packages) when from their point of view one should be sufficient. Am I thinking about that right?

> The pattern of `if` conditions suggests that there isn't much in common either

Are you referring to the `if action == 'start-verification'` block? That's the extra bit for the package claiming case, where we load the packages in question. The flows rejoin just after, in the call to [`start_email_verification`](https://github.com/gratipay/gratipay.com/blob/c3677d4943a4be57dd3f4ecb0aee16187f3c30c7/gratipay/models/participant/email.py#L57-L100), which is where the bulk of the common wiring lives.

> I think it is best to separate these out.
> i.e. use separate functions (and templates) for verifying an email vs verifying linking an email to a package

I agree it's pretty dense (especially when we get to [`finish_email_verification`](https://github.com/gratipay/gratipay.com/blob/c3677d4943a4be57dd3f4ecb0aee16187f3c30c7/gratipay/models/participant/email.py#L218-L254)), but I think it's necessary to support the simpler, single-step user experience during package claiming. Do you see a way to refactor while preserving the single-step flow?